lib/collaborative_filtering.py

```diff
@@ -194,8 +197,17 @@ def get_predictions(self):
         :returns: A (user, document) matrix of predictions
         :rtype: ndarray
         """
-        return self.user_vecs.dot(self.item_vecs.T)
-
+        if self.predictions is None:
+            if self._is_hybrid:
+                ## Train Linear Regression
+                collaborative_predictions = self.user_vecs.dot(self.item_vecs.T)
+                regr = LinearRegression(self.train_data, self.test_data, self.item_based_ratings, collaborative_predictions)
+                self.predictions = regr.train()
+                print("returned linear regression ratings")
+            else:   
+                self.predictions = self.user_vecs.dot(self.item_vecs.T)
```

Comment:
This computation can be removed before the condition, since it's common in both branches of the if/else